### ISSUE TITLE
Data partition size

### ISSUE THREAD

**Issue by JanSykor** [procount/pinn#861]
Hello @procount 
Is there any reason to keep Data partition in Pinn on 500MB? On new installation it was possible to change of size of all systems but not Data partition size.
I bought 512 GB raspberry SSD with M2 hat and want keep rest of disk for data, but because it was not possible to change Data size, I am shortening last project space to 10GB and from rest made new ext4 partition for data. After this I can not boot my both raspios systems. IT stacks on some message about missing partition please sebe attached picture. Do you have some idea how to fix this problem?
Strange is that LibreElec working without problem.
![IMG_20241224_202644](https://github.com/user-attachments/assets/1a6d054d-49e4-4c51-b840-ff1f68a4adc3)
Thanks for your help.
Jan

**Comment by procount** [comment]
By altering the partitions manually, you have probably inadvertently changed the diskID. Some tools like Parted may do this. As you are using an SSD, it will be using partuiid references based on the diskID, so it won't find the partitions. (LibreELEC may use labels or another method to reference its partitions - just a guess - I haven't checked)
Posting the output of `fdisk -l` and the contents of /settings/installed_os.json  or the cmdline.txt of an OS that will not boot will confirm this. (/settings refers to partition No. 5 when mounted in PINN).
Restoring the diskID to that referenced in installed_os.json will probably fix it. (Just ask if you don't know how to do this).

Regarding the data partition size, this can now be adjusted on installation when the Adjust Size of OSes dialog comes up. Even though the data partition is indicated as having a fixed size, this just means any spare capacity will not be shared to it, but you can add any extra size you want. But first you must remove the extra allocations from the other OSes to make room.
I suggest clicking the Clear button, entering the extra MBs you want to add to the Data Partition, then click the Balance button to share out any remaining space among the other expanding partitions. Of course you can manually set the extra sizes for the other OSes instead if you wish.

**Comment by JanSykor** [comment]
Hello @procount 
I am sending requested file in zip - I am not sure what is wrong.
If we did not fix it, I will try your sugestion with Clear and reinstall
Thanks for your help
Jan
[pack.zip](https://github.com/user-attachments/files/18262740/pack.zip)

**Comment by procount** [comment]
The fdisk command in Raspios initramfs is not very comprehensive and does not show the diskID.
Could you repeat it from PINN's command shell?
Boot into PINN (press shift if necessary to remain in PINN)
Ctrl-Alt-F2 to get to the command prompt.
Login with username/password of root/raspberry
Capture the output of `fdisk -l`

Alternatively, if you have enabled PINN's ssh, you can login remotely with a PuTTY and capture the output that way.

**Comment by JanSykor** [comment]
Hello @procount 
Here Is output from pinn
![IMG_20241228_131406](https://github.com/user-attachments/assets/ed70815a-e72a-492a-8769-0fae378d7cb3)
![IMG_20241228_131456](https://github.com/user-attachments/assets/fb8d8e91-c8fc-4c04-96b2-2e68f696be94)

**Comment by procount** [comment]
Yeah, you see the current disk Identifier from fdisk is 0x6d144741, but in the cmdline.txt files, it is expecting it to be 7cde0769. So changing the partition sizes has changed the diskID.

You can change the diskID back again using fdisk.
```
fdisk /dev/nvme0n1
x
i
7cde0769
r
w
q
```

**Comment by JanSykor** [comment]
Yes you have right. Now both system is booting and in pinn fdisk is correct diskID.
Just the name must be in format **0x**7cde0769 and command **w** not working as expected (rereading), but after restart it was ok.
![1735425975499s](https://github.com/user-attachments/assets/de9d7b5e-b339-4ca5-90d4-3ea0eac43266)

Unfortunately lite system cannot see other drives, but I think this is not related to pinn

Anyway thank you very much for your quick help
Jan

**Comment by buvinghausen** [comment]
@procount I'm assuming the preferred workaround is to just allocate less space to each OS leaving free space at the end of the drive which you can then go in and manually create the data partition which presumably wouldn't mess with any of the drive ids?

**Comment by procount** [comment]
No, not really, That's almost what Jan did at the beginning. 
PINN allocates ALL of the disk capacity to the initially installed OSes, so it's not really possible.
Better just to add the data partition at the start and adjust its size in the dialog just before it gets installed.

**Comment by buvinghausen** [comment]
@procount the UI states the data partition is a fixed size disk and won't let you expand it past the 512 gb amount.  I wound up just leaving a bunch of free space and it appears pinn makes a share partition with the balance (wihch was ultimately what I was after)

**Comment by procount** [comment]
Please see the post above: https://github.com/procount/pinn/issues/861#issuecomment-2562044478

> Regarding the data partition size, this can now be adjusted on installation when the Adjust Size of OSes dialog comes up. Even though the data partition is indicated as having a fixed size, this just means any spare capacity will not be shared to it, but you can add any extra size you want. But first you must remove the extra allocations from the other OSes to make room.
I suggest clicking the Clear button, entering the extra MBs you want to add to the Data Partition, then click the Balance button to share out any remaining space among the other expanding partitions. Of course you can manually set the extra sizes for the other OSes instead if you wish.

But if you've managed it another way, then that's fine. Note that PINN won't create additional partitions. It just shares the spare capacity out amongst the dynamic partitions of all the OSes.
